src/middlewares/query.check.js

```diff
@@ -0,0 +1,54 @@
+import { Op } from 'sequelize';
+
+export const searchArticleQuerybuilder = (query) => {
+  const validQueries = {
+    author: 'authorId',
+    keyword: 'description',
+    tag: 'taglist',
+    title: 'title'
+  };
+  let queries = {};
+
+  Object.keys(query).forEach((key) => {
+    queries = { ...queries, [validQueries[key]]: query[key] };
+  });
+  // wildcard search
+  if (queries.description) {
+    queries.description = {
+      [Op.iLike]: `%${queries.description}%`
+    };
+  }
+  if (queries.title) {
+    queries.title = {
+      [Op.iLike]: `%${queries.title}%`
+    };
+  }
+
+  return queries;
+};
+
+export const checkQuery = (req, res, next) => {
+  const { query } = req;
+  const validQueries = ['author', 'keyword', 'tag', 'title', 'offset', 'limit'];
```

Comment:
consider the other Queries might appear like `newest ` like `weekly` as we had that query in the previous UI we discussed.